extensions/mypy_extensions.py

```diff
@@ -7,16 +7,76 @@
 
 # NOTE: This module must support Python 2.7 in addition to Python 3.x
 
+import sys
+import types
+from typing import _type_check  # type: ignore
```

Comment:
OK, it's fine to keep, but please add a comment to the code for posterity. (I also worry that before we know it this will become a de-facto public API -- I've seen others ask for the same functionality, e.g. in the issue asking to restore get_type_hint() for Python 2.)